### DISCUSSION TITLE
How can I get my yeast model grow under anaerobic conditions?

### DISCUSSION THREAD

**Discussion by thuynh842** [opencobra/cobrapy#1131] [Q&A]
For my undergraduate thesis, I am working with the iMM904 yeast model. However, whenever I try to run FBA under anaerobic conditions (by either setting the EX_o2_e in medium to 0.0 or knocking out the EX_o2_e reaction), the biomass optimize function always returns 0. How do I fix this problem?

**Comment by Midnighter** [comment]
Hi @thuynh842,

There are a few possible reasons, like too high enforced ATP maintenance flux or other problems with exchange reactions. Before you dig any deeper, I recommend that you try the yeast model that is continuously updated by the systems and synthetic biology groups at Chalmers https://github.com/SysBioChalmers/yeast-GEM/releases/. The IMM904 model is quite old by now.

**Comment by danolson1** [comment]
Try setting the the oxygen exchange flux very low instead of zero. Also, make sure your model can produce ethanol.

-Dan

**Reply by thuynh842** [reply]
Hi @danolson1, thank you for your response. My model still produces ethanol (though at a much smaller quantity due to smaller glucose uptake rate).

**Reply by lmigueel** [reply]
Hi! Did you solve the problem? I have the same issue as you for anaerobic simulation using iMM904.

**Comment by bdelepine** [comment]
Hi everyone,

Not getting any growth in a strict anaerobic medium with a yeast model is expected. Indeed, experimentalists need to provide additional substrates for growth to occur in strict anaerobiosis. Once those substrates are supplemented to your model, you should have growth.

In [this study](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2507709/) for instance, the authors state:
> While the high fermentative capacity enables _S. cerevisiae_ to produce energy at a sufficient rate even in anaerobic conditions [10], constant anaerobic growth requires addition of unsaturated fatty acids and ergosterol to the culture medium since oxygen is an essential reactant in sterol biosynthesis and anabolic desaturation reactions [1,11].

So, for iMM904, this should work:
```python
m = cobra.io.web.load_model("iMM904")

m.reactions.EX_o2_e.bounds = (0, 1000)

supplements  =[
 'EX_ergst_e',
 'EX_zymst_e',
 'EX_hdcea_e',
 'EX_ocdcea_e',
 'EX_ocdcya_e',
]

for rid in supplements:
    reac = m.reactions.get_by_id(rid)
    reac.bounds = (-1, 1000)

s = m.optimize()
print(m.summary(s))
```

Note that, alternatively, you may use pseudo anaerobic conditions [like it was suggested in another answer](https://github.com/opencobra/cobrapy/discussions/1131#discussioncomment-1644954). In [this study](https://pubmed.ncbi.nlm.nih.gov/15355549/) for instance the authors state that:
>"The oxygen uptake rate was not allowed to reach zero because anaerobic simulations required additional supplements to maintain cell growth (ergosterol and zymosterol)."


HTH
